src/routes/posts.js

```diff
@@ -36,7 +36,10 @@ const fetchPosts = catchAsync( async(req, res) => {
 				profilePictureUrl : post.profilePictureUrl,
 				email : post.email
 			})})
-		)
+		),
+		metadata: {
+			totalCount: feed.length
```

Comment:
So the aim of the `totalCount` is to give an idea of the total number of posts in the db, so that the client can know if there is more documents to fetch, so its not really about counting the number of documents being returned.

It involves having a count query to the database. 

More information on the reasoning is highlighted in the technical and additional considerations of the issue description